test/distributed/fsdp/test_fsdp_misc.py

```diff
@@ -51,111 +50,121 @@ def process_group(self):
 
     @skip_if_lt_x_gpu(2)
     def test_device_id_auto_wrap(self):
-        """
-        Test auto wrapping propagates the device id.
-        """
-        model = TransformerWithSharedParams(group=self.process_group)
-        my_auto_wrap_policy = functools.partial(
+        """Tests that ``auto_wrap_policy`` propagates ``device_id`` to all
+        nested FSDP instances."""
+        auto_wrap_policy = functools.partial(
             transformer_auto_wrap_policy,
-            transformer_layer_cls={TransformerEncoderLayer, TransformerDecoderLayer}
+            transformer_layer_cls={TransformerEncoderLayer, TransformerDecoderLayer},
         )
-        wrapped = FSDP(
-            model,
-            auto_wrap_policy=my_auto_wrap_policy,
-            device_id=torch.cuda.current_device()
+        fsdp_kwargs = {
+            "auto_wrap_policy": auto_wrap_policy,
+            "device_id": torch.cuda.current_device(),
+        }
+        fsdp_model = TransformerWithSharedParams.init(
+            self.process_group,
+            FSDPInitMode.RECURSIVE,
+            CUDAInitMode.CUDA_BEFORE,
+            fsdp_kwargs,
         )
-        # All FSDP instances should have device_id set
-        for m in FSDP.fsdp_modules(wrapped):
-            self.assertEqual(m.device_id, torch.device("cuda", torch.cuda.current_device()))
+        for fsdp_module in FSDP.fsdp_modules(fsdp_model):
+            self.assertEqual(
+                fsdp_module.device_id,
+                torch.device("cuda", torch.cuda.current_device()),
+            )
 
     @skip_if_lt_x_gpu(2)
     @parametrize("use_index", [True, False])
     def test_fsdp_device_id(self, use_index):
         """
-        If CPU module is passed into FSDP with device_id
-        argument, it is moved to the GPU with that device_id.
+        Tests the FSDP ``device_id`` argument:
+          - Wrapping a CPU module should move the module to the GPU matching
+          ``device_id``
+          - Wrapping a GPU module already on the GPU matching ``device_id``
+          should not raise an error
+          - Wrapping a GPU module already on GPU and passing a GPU device
+          without specifying a device ID (i.e. ``torch.device("cuda")``) warns
         """
         dev_id = (
             torch.cuda.current_device() if use_index
             else torch.device("cuda", torch.cuda.current_device())
         )
 
-        def _check_device_matches(fsdp, dev_id):
-            devices = {p.device for p in fsdp.parameters()}
+        def _check_device_matches(module, device_id):
```

Comment:
Changed this argument to `device_id` just to prevent accidental variable shadowing.